### DISCUSSION TITLE
Newly Installed Homebrew on Mac - Showing 2 Git Versions

### DISCUSSION THREAD

**Discussion by jmmb** [Homebrew/discussions#3921] [Getting started]
Hello, I've followed the instructions at https://jekyllrb.com/docs/installation/macos/ to install Jekyll on my Intel Mac (w/ latest Monterey MacOS vsn.), and in the process installed Homebrew.

And before going further, it was recommended to run `brew doctor` to see if everything was looking okay.  There are a couple (hopefully) irrelevant warnings, but the following seems fairly important:

Warning: An outdated version (2.10.1) of Git was detected in your PATH. Git 2.14.3 or newer is required for Homebrew.
Please upgrade:
  brew upgrade git

I did try addressing the third warning by running: `brew upgrade git`
However, afterward when I run brew doctor again, I still see the outdated message. And when I do `git --version`, it returns 2.10.1.  So it appears that`brew upgrade git` isn't working?

If you can give any advice, it'd be much appreciated.

Thanks,
Jim

**Comment by SMillerDev** [comment]
`brew doctor` also mentions that if you don't have issues there is no need to worry about the output. If you want to update your git you might want to check for Xcode or command line tools updates too.

**Comment by jmmb** [comment]
Thanks for the reply, @SMillerDev. I agree I can probably ignore the first two warnings, but I do want to upgrade the Git version. BTW I also took your advice to check the Xcode and command line tools versions, and those are all up to date.

So I've been following the guide at https://ajahne.github.io/blog/tools/2018/06/11/how-to-upgrade-git-mac.html, and it seems that I'm running the official Git distribution as opposed to the Apple version of Git.

So with that said, here is the status:

`git --version` yields: git version 2.10.1
`brew doctor` (still) says: An outdated version (2.10.1) of Git was detected in your PATH.
Git 2.14.3 or newer is required for Homebrew ...

But then when I run `brew upgrade git`, the result is: Warning: git 2.38.1 already installed

So there is some conflict going on here, and I really feel I need to get that figured out.  Any additional thoughts from anyone will be much appreciated.

**Comment by jmmb** [comment]
Update, I found this, which seems to be my problem as well: https://stackoverflow.com/questions/63985000/git-version-displays-wrong-version-on-terminal

On that advice, I typed `brew install git`
That returned: Warning: git 2.38.1 is already installed, it's just not linked.

So I typed:  `brew link git`
And what is outputted is the following:

Linking /usr/local/Cellar/git/2.38.1... 
Error: Could not symlink bin/git
Target /usr/local/bin/git
already exists. You may want to remove it:
  rm '/usr/local/bin/git'

To force the link and overwrite all conflicting files:
  brew link --overwrite git

To list all files that would be deleted:
  brew link --overwrite --dry-run git

So with these options, I'm not sure what to do... but looking at https://stackoverflow.com/questions/66481376/updated-git-on-mac-old-version-still-showing, it appears this is the solution:  `brew link --overwrite git`

However, I don't want to do anything further until I'm more confident of the correct next step(s) ... Again, some advice will be much appreciated!

**Reply by jmmb** [reply]
Update: well, I couldn't wait any longer, so after more searching it seemed like the consensus was to do this:

`brew link --overwrite git`

I did this, and it outputted: Linking /usr/local/Cellar/git/2.38.1... 228 symlinks created.

I then did `git --version`, and the result now is: git version 2.38.1

And when I run brew doctor again, I see that got rid of the Git warning as well as one of the other warnings I was initially seeing.  So it looks like this fixed it, in case it helps anyone else.

**Reply by ainzizou** [reply]
this solution `brew link --overwrite git` works for me, thanks @jmmb !
